File: backend/lumen_schemas/stock_music.py
"""M36.2.2: Stock background-music Pydantic schemas.

Mirrors ``lumen_schemas.stock_asset`` — separate Create/Read/ListItem
shapes so the gallery list stays light while the detail page surfaces
every column. The audio proxy at ``/api/v1/stock-musics/{id}/file`` is
Bearer-protected; consumers must use ``fetch + Bearer + blob +
createObjectURL`` (see MEMORY 2026-06-20).
"""
from datetime import datetime
from typing import List, Literal, Optional

from pydantic import BaseModel, ConfigDict


StockMusicSource = Literal["builtin", "uploaded"]


class StockMusicListItem(BaseModel):
    """Lightweight row for the gallery list — name + duration + category."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    category: str
    description: Optional[str] = None
    mime_type: str
    file_size: int
    duration_seconds: float
    source: StockMusicSource
    created_at: datetime


class StockMusicDetail(StockMusicListItem):
    """Full row returned by ``GET /stock-musics/{id}``.

    Includes ``file_path`` so the frontend can build the proxy URL
    (``/api/v1/stock-musics/{id}/file``). The proxy enforces Bearer
    auth — we do NOT leak an absolute file path.
    """

    file_path: str
    tenant_id: Optional[int] = None

File: frontend/types/stock-music.ts
// frontend/types/stock-music.ts
// M36.2.2 — stock background-music library
//
// 1:1 mirror of backend lumen_schemas/stock_music.py. The proxy endpoint
// returns a FileResponse that requires Bearer auth, so the frontend must
// consume the audio bytes via `fetch + blob + createObjectURL` (see
// MEMORY 2026-06-20). For composition the service only needs the id
// (resolved by the backend's _resolve_asset_to_path), but the audio
// preview UI needs the full proxy URL.

export type StockMusicSource = "builtin" | "uploaded";

/** Row returned by GET /api/v1/stock-musics/ (list endpoint). */
export interface StockMusicListItem {
  id: number;
  name: string;
  category: string;
  description: string | null;
  mime_type: string;
  file_size: number;
  duration_seconds: number;
  source: StockMusicSource;
  created_at: string;
}

/** Full row returned by GET /api/v1/stock-musics/{id}. */
export interface StockMusicDetail extends StockMusicListItem {
  file_path: string;
  tenant_id: number | null;
}

/** Query params for GET /api/v1/stock-musics/. */
export interface StockMusicListParams {
  page?: number;
  page_size?: number;
  category?: string;
  search?: string;
}

/** Flat object returned by listStockMusics() — service unwraps PaginatedResponse. */
export interface StockMusicListResult {
  items: StockMusicListItem[];
  total: number;
  page: number;
  page_size: number;
}
